### PR TITLE
MAINT: Fix clang build and remove some unicode characters

### PR DESCRIPTION
**Pull request by peterbell10**
When compiling with clang 10 or 11, I get errors in `biasedurn/wnchyppr.cpp`:
```
scipy/stats/biasedurn/wnchyppr.cpp:553:20: error: non-constant-expression cannot be narrowed from type 'int32_t' (aka 'int') to 'double' in initializer list [-Wc++11-narrowing]
   double xx[2] = {x, n-x};
```
On macOS, where clang is the default, this was fixed by adding a command line flag `-Wno-narrowing`. This PR takes the other approach of adding the appropriate `static_cast`s.

Also, when editing this file I get invalid unicode character errors. So replaced those with plain ascii.

### CODE REVIEW

**Review by rgommers — APPROVED** [review]
Thanks Peter. LGTM - the extra compile flags approach isn't portable indeed. I'll defer to @mckib2 for the rest.

**Review by mckib2 — APPROVED** [review]

**Comment by tylerjereddy** [comment]
Alright, two core dev approvals and CI fails don't look related.

**Comment by tylerjereddy** [comment]
Thanks @peterbell10 et al.

**Pull request merged**
